### PR TITLE
snippets: dyld: add image dependency resolving

### PR DESCRIPTION
**Pull request by yanivhasbani**
snippets: dyld: added image dependency resolving

### CODE REVIEW

**Comment by doronz88** [comment]
rename commit message to command form - should be:
```
snippets: dyld: add image dependency resolving
```

File: hilda/snippets/macho/all_image_infos.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    def image_dependencies(self, image_name):
+        images = [image for image in self.find_images(image_name)]
+
+        if len(images) == 0:
+            self.reload()
+            images = [image for image in self.find_images(image_name)]
+
+            if len(images) == 0:
+                print(f"Image {image_name} is not loaded")
```
Comment (doronz88) [inline hilda/snippets/macho/all_image_infos.py:78]: use either `logging` module for your own module logging

File: hilda/snippets/macho/all_image_infos.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        image = unique_images[0]
+        if len(unique_images) > 1:
+            image = self.__select_specific_image(unique_images)
+
+        dependencies = set([image_name.path for image_name in image.load_commands.dylib_commands])
+
+        return dependencies
+
+    def __select_specific_image(self, images):
+        while 1:
```
Comment (doronz88) [inline hilda/snippets/macho/all_image_infos.py:99]: replace this input logic with `humanfriendly`:
https://humanfriendly.readthedocs.io/en/latest/api.html#humanfriendly.prompts.prompt_for_choice

File: hilda/snippets/macho/all_image_infos.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        return self.__image_infos
+
+    def find_images(self, name):
+        return [image for image in self.images if name in image.file_path]
+
+    def image_dependencies(self, image_name):
+        images = [image for image in self.find_images(image_name)]
+
+        if len(images) == 0:
+            self.reload()
```
Comment (doronz88) [inline hilda/snippets/macho/all_image_infos.py:74]: does this operation take a long time? why else performing the `images = [image for image in self.find_images(image_name)]` twice?

if so, add a comment

File: hilda/snippets/macho/all_image_infos.py (hunk 1 above)
Comment (doronz88) [reply hilda/snippets/macho/all_image_infos.py:78]: Also, i'm wondering.. Isn't raising an exception the user can catch better?

**Review by doronz88 — CHANGES_REQUESTED** [review]
don't forget to fix the commit message

File: hilda/snippets/macho/all_image_infos.py (hunk 1 above)
Comment (yanivhasbani) [reply hilda/snippets/macho/all_image_infos.py:78]: They way I see it, the function is responsible to validate the input and give the user the option to fix that.
Raising an exception goes against that assumption. 

What do you think?

File: hilda/snippets/macho/all_image_infos.py (hunk 3 above)
Comment (yanivhasbani) [reply hilda/snippets/macho/all_image_infos.py:74]: Added

File: hilda/snippets/macho/all_image_infos.py (hunk 1 above)
Comment (doronz88) [reply hilda/snippets/macho/all_image_infos.py:84]: ok

File: hilda/snippets/macho/all_image_infos.py (hunk 3 above)
Comment (doronz88) [reply hilda/snippets/macho/all_image_infos.py:80]: `could`

File: hilda/snippets/macho/all_image_infos.py (hunk 2 above)
Comment (doronz88) [reply hilda/snippets/macho/all_image_infos.py:110]: no need for the `while` look. the `prompt_for_choice` does that for you

**Review by doronz88 — CHANGES_REQUESTED** [review]
minor fixes

File: hilda/snippets/macho/all_image_infos.py (hunk 3 above)
Comment (doronz88) [reply hilda/snippets/macho/all_image_infos.py:80]: still exists

File: hilda/snippets/macho/all_image_infos.py (hunk 2 above)
Comment (yanivhasbani) [reply hilda/snippets/macho/all_image_infos.py:107]: `prompt_for_choice` may throw an exception though..

If that is the case, I wish to give the user another attempt to provide a valid input

File: hilda/snippets/macho/all_image_infos.py (hunk 3 above)
Comment (yanivhasbani) [reply hilda/snippets/macho/all_image_infos.py:80]: Missed that one 🤦

Comment (yanivhasbani) [reply hilda/snippets/macho/all_image_infos.py:80]: Done

**Pull request merged**